### PR TITLE
Add GitHub Actions Specs workflow

### PR DESCRIPTION
**Pull request by igor-makarov**
Travis is slow, this is a full Core test build running in GH Actions.

### CODE REVIEW

**Comment by dnkoutso** [comment]
@igor-makarov can you point to `1-10-stable`?

**Comment by igor-makarov** [comment]
@dnkoutso Done!

File: .github/workflows/Specs.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  specs:
+    strategy:
+      matrix:
+        os: [ubuntu-16.04]
+        ruby: [2.3.3, 2.4, 2.5, 2.6, 2.7]
+        include:
+          - os: macos-10.15
+            ruby: system
+
+    name: Ruby ${{ matrix.ruby }}
```
Comment (endocrimes) [inline .github/workflows/Specs.yml:13]: Can you include the OS in the workflow name? Makes it much easier to skim

File: .github/workflows/Specs.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        run: |
+          gem install bundler -v "~> 1.17"
+          bundle install --jobs 4 --retry 3 --without debugging documentation
+
+      - name: Run Specs + Rubocop
+        run: bundle exec rake spec
+
+on:
+  push:
+    branches:
```
Comment (endocrimes) [inline .github/workflows/Specs.yml:41]: Is there a reason to not run on all branches? (Getting a ci run before opening the PR itself can be nice sometimes)

**Review by endocrimes — COMMENTED** [review]
Thanks for doing this!

File: .github/workflows/Specs.yml (hunk 1 above)
Comment (igor-makarov) [reply .github/workflows/Specs.yml:13]: @endocrimes - I can't include just the OS type, and I figured that saying `ubuntu-16.04` is a bit verbose. Do you think it's better to include it, even if verbose?

File: .github/workflows/Specs.yml (hunk 2 above)
Comment (igor-makarov) [reply .github/workflows/Specs.yml:41]: @endocrimes It used to be the case that this caused a weird duplication in test runs that would show up in checks. I'll try and see if it's still the case.

Comment (igor-makarov) [reply .github/workflows/Specs.yml:41]: @endocrimes Yup, it did that. Reverting 😭

File: .github/workflows/Specs.yml (hunk 1 above)
Comment (igor-makarov) [reply .github/workflows/Specs.yml]: @endocrimes I have added the runner name to the job title - looks a bit verbose (e.g. `Specs / ubuntu-16.04 / Ruby 2.6 (pull_request)`) but livable.

Comment (endocrimes) [reply .github/workflows/Specs.yml]: Verbose is good for things like this I think 😅  - Thanks for adding it!

**Pull request merged**